CONTRIBUTING.md

```diff
@@ -0,0 +1,73 @@
+
+* Mixing whitespace changes with functional code changes.
+* Mixing two unrelated functional changes in the same commit.
+* Sending large new features in a single commit.
+
+Expected git commit message structure (Based on [this article](https://chris.beams.io/posts/git-commit/)):
+
+* The first line should ideally be limited to 50 characters, but must be no more than 70 characters.
+* The first line should not end with a full stop and the first word should be capitalised.
+* Use the imperative mood in the subject line - e.g. ```Fix a typo in CONTRIBUTING.md```, ```Remove if/else block in myfile.sh```, ```Eat your dinner```.
+* Insert a single blank line after the first line.
+* Subsequent lines should be wrapped at 72 characters.
+* Provide a detailed description of the change in the following lines, using the guidelines in the section below.
+* The last line of the commit should be a reference to the issue being fixed using the word JIRA, followed by the JIRA issue Key e.g. ```JIRA: HPCDIR-9999```
+
+In your commit message please consider the following points:
+
+* In the commit subject line, provide a brief description of the change.
+* The commit message must be descriptive and contain all the information required to fully understand and review the patch.
+* Do not assume the reviewer understands what the original problem was.
+* Do not assume the reviewer has access to external web services or site.
+* Do not assume the code is self-evident or self-documenting.
+* Describe why the change is being made.
+* Ensure sufficient information is provided to review your commit.
+* Each commit should ideally solve a single problem. Where there are several things to fix to reach the goal identified by the card, multiple commits should be used. Each commit should be as easy as possible to consume (or revert) as a single, complete change.
+
+### Pull requests
+
+* Pull Requests (PRs) should be made from a development branch in the main repo. 
+* A PR should contain a single commit, or multiple commits all related to the problem being solved by the PR.
+* The PR title and message should contain information relevant to why the commit(s) are happening. This can be based off the commit message.
+* As with commit messages, the PR description should reference the original issue.
+
+#### Reviewing pull requests
+
+When reviewing a PR, please ensure the included commit(s):
+
+* Actually works to fix the issue.
+* Passes any tests that are configured to run.
+* Contains a single commit, or multiple commits all related to the problem being solved by the PR.
+* Does not overreach. Each PR should be self contained to only address the issue at hand (see above).
+
+#### Merging pull requests
+
+In order for a PR to be merged, the following criteria should be met:
+
+* Any configured tests must have passed.
+* There must be at least 2 members of the engineering team review and give the PR a +1 approval using the review guidelines above.
```

Comment:
Controversial. 

There's only 2 of us in platform development! :-D

Reply:
there we go, we already have justification for a new head :D

oooooor, we can just add a comment that if the team is only 2, then a single +1 is acceptable. This is enforcable in github with protected branches - not sure about gitlab - so we'd need to set your repos accordingly.

Reply:
This is likely to reduce our velocity as it takes a fair amount of effort to review changes and requires 3 people in a team of 4, which when someone is on holiday is the whole team for platform engineering even.  

Can we reduce this to 1 for now.

Reply:
I'd say the number of required reviewers should depend on the team size. With five or six committers in a team, it starts to make sense to have two reviewers, but there's a balance to be struck between the quality code review and the speed of development, which is also quite important.

Reply:
I agree with Darryl - while having 2 reviewers is good practice, I've never seen it work like that in a small team and we should be honest with ourselves: if it is unrealistic, then there's no point making a rule for it that we'll never live up to. Best to put at "least one, ideally more where possible" or something

Reply:
shall we make this 1 for all repos for now then, since that also satisfies the 2-person dev team